### ISSUE TITLE
Reschedule Functionality Has Highly Variable Performance

### ISSUE THREAD

**Issue by therabidbanana** [hired/reactor#82]
Based on how many items might be in the scheduled set for Sidekiq, scanning over them all can end up being extremely expensive. We see average times of 3+ seconds in web requests using the `publishes .... at: ` feature that automatically reschedules.

<img width="979" alt="screen shot 2018-05-07 at 11 16 52 am" src="https://user-images.githubusercontent.com/18348/39717502-516c60e0-51e8-11e8-9c03-8502afa6242a.png">

In short, we use functionality documented here: https://github.com/mperham/sidekiq/wiki/API , but ignored this warning:

> You should not be scanning through queues/sets and deleting jobs unless something has gone wrong and you need to repair data manually.

Maintaining the functionality by allowing duplicate self-destructing scheduled jobs might be possible and avoid the performance issues.

**Comment by therabidbanana** [comment]
Likely instead of having a hook for rescheduling:

https://github.com/hired/reactor/blob/f41266ac5b8c26f26bd2a113845340b292a1f653/lib/reactor/models/concerns/publishable.rb#L31-L47

We would make it so scheduled events could abort by updating the `need_to_fire` logic to care about scheduling as well as the generic "ifarg" logic:

https://github.com/hired/reactor/blob/8c68156554458b2e8d0bac2f60c82ff7c14ae3f7/lib/reactor/event.rb#L24-L37

In theory, if we queue a job with Reactor data including the attribute we're watching to schedule and current value, then we can check the value is still equal to the most recent value when the actor gets hydrated: https://github.com/hired/reactor/blob/8c68156554458b2e8d0bac2f60c82ff7c14ae3f7/lib/reactor/event.rb#L25

In publishable, we use the `:at` param and call that method to get the value - we could save `:at_attribute` as the actual attribute so we can compute this check on 
https://github.com/hired/reactor/blob/f41266ac5b8c26f26bd2a113845340b292a1f653/lib/reactor/models/concerns/publishable.rb#L72
